### DISCUSSION TITLE
Solving a nonlinear 1-D Poisson equation

### DISCUSSION THREAD

**Discussion by MPBora** [zwicker-group/py-pde#235] [Q&A]
Any clue on how to define the fields and which solver to use for a nonlinear Poisson equation in 1-D? The equation in question is a simple Poisson equation appearing in nonlinear Plasma
`d^2 f/dx^2=e^f-(1-a f)`
where `a` is a constant. The equation is to be solved in a Cartesian domain `[0,b]` with the BCs : `f'(0)=0, f(b)=0`. Any help will be appreciated.

**Comment by david-zwicker** [comment]
I'm afraid that the package does not offer any methods to solve this equation directly since the package mainly focuses on PDEs originating from dynamical systems. However, you might be able to convert your problem to one which relaxes toward the desired solution, e.g., simulate
```
df/dt = d^2 f/dx^2 - e^f + (1 - a f)
```
The stationary state solution of this equation should correspond to the solution of your original equation.

You can implement the equation using the [`PDE` class](https://py-pde.readthedocs.io/en/latest/packages/pde.pdes.pde.html), e.g., like in [this example](https://py-pde.readthedocs.io/en/latest/examples_gallery/pde_1d_expression.html#sphx-glr-examples-gallery-pde-1d-expression-py).

**Reply by MPBora** [reply]
I can understand that. The example is fine. But how to set up the boundary conditions? Can you provide an example?

**Reply by david-zwicker** [reply]
Your particular conditions can be set as `PDE(..., bc=[{'derivative': 0}, {'value': 0}])`. This is explained in more detail in the documentation.
